src/irmin-pack/unix/control_file.ml

```diff
@@ -135,19 +247,10 @@ module Make (Io : Io.S) = struct
     let open Result_syntax in
     let* string = Io.read_all_to_string io in
     (* Since the control file is expected to fit in a page,
-       [read_all_to_string] should be atomic for most filesystems.
-
-       If [string] is larger than a page, it either means that the file can be
-       corrupted or that the major version is not supported. Either way it will
-       be detected by [Data.of_bin_string] or have an invalid checksum. *)
-    let* payload = Data.of_bin_string string in
-    match payload with
-    | V3 payload -> Ok (upgrade_v3_to_v4 payload)
-    | V4 payload ->
-        if checksum_is_valid payload then Ok payload
-        else Error `Corrupted_control_file
-
-  let create_rw ~path ~overwrite payload =
+       [read_all_to_string] should be atomic for most filesystems. *)
+    Serde.of_bin_string string
```

Comment:
Not for this PR, but we should just write to a temporary file and rename it to get that sweet posix atomic update!

Reply:
And seems like it was always meant to be a temporary thing? 

Looking at the [comment above](https://github.com/mirage/irmin/pull/2180/files/7b9e971b221b577c26dd533f9f8853bf77967325#diff-74307791c5ce4d5c06405d7aab601479f53bf531e3d3b3833880157e58c2b079R239-R241):
> The data must fit inside a single page for atomic updates of the file.
       This is only true for some file systems. This system will have to be
       reworked for [V4].

Opened an issue to track https://github.com/mirage/irmin/issues/2181